Tests/SubprocessTests/ProcessMonitoringTests.swift

```diff
@@ -0,0 +1,361 @@
+        let devnullFd: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        let devnull = HANDLE(bitPattern: _get_osfhandle(devnullFd.rawValue))!
+        #else
+        let devnull: FileDescriptor = try .openDevNull(withAccessMode: .readOnly)
+        #endif
+        return CreatedPipe(
+            readFileDescriptor: .init(devnull, closeWhenDone: true),
+            writeFileDescriptor: nil
+        )
+    }
+
+    private func devNullOutputPipe() throws -> CreatedPipe {
+        #if os(Windows)
+        let devnullFd: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        let devnull = HANDLE(bitPattern: _get_osfhandle(devnullFd.rawValue))!
+        #else
+        let devnull: FileDescriptor = try .openDevNull(withAccessMode: .writeOnly)
+        #endif
+        return CreatedPipe(
+            readFileDescriptor: nil,
+            writeFileDescriptor: .init(devnull, closeWhenDone: true)
+        )
+    }
+}
+
+// MARK: - Basic Functionality Tests
+extension SubprocessProcessMonitoringTests {
+    @Test func testNormalExit() async throws {
+        let config = self.immediateExitProcess(withExitCode: 0)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+        let monitorResult = try await monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+
+        #expect(monitorResult.isSuccess)
+    }
+
+    @Test func testExitCode() async throws {
+        let config = self.immediateExitProcess(withExitCode: 42)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+        let monitorResult = try await monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+
+        #expect(monitorResult == .exited(42))
+    }
+
+    #if !os(Windows)
+    @Test func testExitViaSignal() async throws {
+        let config = Configuration(
+            executable: .path("/usr/bin/tail"),
+            arguments: ["-f", "/dev/null"]
+        )
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+
+        async let monitorResult = try monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+        // Send signal to process
+        try spawnResult.execution.send(signal: .terminate)
```

Comment:
Theoretically racy on Darwin and OpenBSD (or any other unix without process descriptors).

By the time send() is called, monitorProcessTermination might've reaped the PID. It looks like there's no reason you can't just call send before monitorProcessTermination instead.